node-notes-cli/app.js

```diff
@@ -0,0 +1,41 @@
+import * as fs from 'node:fs';
+import * as fsPromises from 'node:fs/promises';
+
+const fileData = fs.readFileSync('data.json', 'utf8');
+const data = JSON.parse(fileData);
+const action = process.argv[2];
+
+switch (action) {
+  case 'read':
+    for (const key in data.notes) {
+      console.log(`${key}: ${data.notes[key]}`);
+    }
+    break;
+  case 'create':
+    data.notes[data.nextId] = process.argv[3];
+    data.nextId++;
+    fsPromises.writeFile('data.json', JSON.stringify(data, null, 2));
```

Comment:
You write this several times. It would be nice to not repeat yourself. Also, it would be nice to catch and report any write errors.